fastly/custom_tls_configuration.go

```diff
@@ -0,0 +1,153 @@
+type CustomTLSConfiguration struct {
+	ID            string       `jsonapi:"primary,tls_configuration"`
+	DNSRecords    []*DNSRecord `jsonapi:"relation,dns_records"`
+	Bulk          bool         `jsonapi:"attr,bulk"`
+	Default       bool         `jsonapi:"attr,default"`
+	HTTPProtocols []string     `jsonapi:"attr,http_protocols"`
+	Name          string       `jsonapi:"attr,name"`
+	TLSProtocols  []string     `jsonapi:"attr,tls_protocols"`
+	CreatedAt     *time.Time   `jsonapi:"attr,created_at,iso8601"`
+	UpdatedAt     *time.Time   `jsonapi:"attr,updated_at,iso8601"`
+}
+
+// DNSRecord is a child of CustomTLSConfiguration
+type DNSRecord struct {
+	ID         string `jsonapi:"primary,dns_record"`
+	RecordType string `jsonapi:"attr,record_type"`
+	Region     string `jsonapi:"attr,region"`
+}
+
+// ListCustomTLSConfigurationsInput is used as input to the ListCustomTLSConfigurationsInput function.
+type ListCustomTLSConfigurationsInput struct {
+	FilterBulk *bool   // Whether or not to only include bulk=true configurations
+	Include    *string // Include related objects. Optional, comma-separated values. Permitted values: dns_records.
+	PageNumber *uint   // The page index for pagination.
+	PageSize   *uint   // The number of keys per page.
+
+}
+
+// formatFilters converts user input into query parameters for filtering.
+func (i *ListCustomTLSConfigurationsInput) formatFilters() map[string]string {
+	result := map[string]string{}
+	pairings := map[string]interface{}{
+		"filter[bulk]": i.FilterBulk,
+		"include":      i.Include,
+		"page[size]":   i.PageSize,
+		"page[number]": i.PageNumber,
+	}
+	for key, value := range pairings {
+		if !reflect.ValueOf(value).IsNil() {
+			result[key] = fmt.Sprintf("%v", reflect.ValueOf(value).Elem())
+		}
+	}
+	return result
+}
+
+// ListCustomTLSConfigurations list all TLS configurations.
+func (c *Client) ListCustomTLSConfigurations(i *ListCustomTLSConfigurationsInput) ([]*CustomTLSConfiguration, error) {
+	p := "/tls/configurations"
+	filters := &RequestOptions{
+		Params: i.formatFilters(),
+		Headers: map[string]string{
+			"Accept": "application/vnd.api+json", // this is required otherwise the filters don't work
+		},
+	}
+
+	r, err := c.Get(p, filters)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := jsonapi.UnmarshalManyPayload(r.Body, reflect.TypeOf(new(CustomTLSConfiguration)))
+	if err != nil {
+		return nil, err
+	}
+
+	con := make([]*CustomTLSConfiguration, len(data))
+	for i := range data {
+		typed, ok := data[i].(*CustomTLSConfiguration)
+		if !ok {
+			return nil, fmt.Errorf("unexpected response type: %T", data[i])
+		}
+		con[i] = typed
+	}
+
+	return con, nil
+}
+
+// GetCustomTLSConfigurationInput is used as input to the GetCustomTLSConfiguration function.
+type GetCustomTLSConfigurationInput struct {
+	ID      string
+	Include *string // Include related objects. Optional, comma-separated values. Permitted values: dns_records.
```

Comment:
I might just be confused 🙂  but could this probably just be a non-pointer string? due to the fact this isn't a write/update operation. 

e.g. we're typically using pointers for basic types that are optional fields because of the problem inherent to a marshal of the data when the field is purposely given a value that aligns with the field type's zero value. 

But that's not relevant in this case I think because it's just a READ operation.

Reply:
That makes sense. Updated.

Reply:
I think you might need to re-push as I can still see a pointer reference used.